### DISCUSSION TITLE
Timeout in Action

### DISCUSSION THREAD

**Discussion by nikkioon** [vuejs/pinia#1305] [Help and Questions]
Hi im working on a Nuxt 3 Project.
Im using a Timeout in my Pinia Action like the Example:

```
exampleFunction() {
      this.ExampleTimeoutVariable = setTimeout(function () {
	      //custom code
      }.bind(this), 500);
}
```
When i write the Action like that i get a h3 error 
<img width="687" alt="2022-05-18 um 13 42 21" src="https://user-images.githubusercontent.com/104486336/169031001-5d122bbe-3bcf-4a6a-81dd-6fc84a5fc837.png">

Is anything wrong with my way? Is there another way to use Timeout and Interval in Pinia?

**Comment by posva** [comment]
you need to make sure the action is awaited in your server wherever the action is called:


```ts
const delay = t => new Promise(r => setTimeout(r, t))
async myAction() {
  await delay(500)
  // custom code
}

// somewhere
await store.myAction()
```

**Comment by nikkioon** [comment]
Thx for the response but i dont think thats the Problem.
I dont call the action from a Server (no SSR) only from client with no return value.
I'm not sure if the error comes from Pinia at all because it only occurs sometimes and I haven't found the exact trigger yet. Maybe it is also an internal Nuxt 3 error after all the beta phase is not over yet. 

Thanks again for the help

**Reply by posva** [reply]
as you can see in the error, it does come from the server, not the client. If you don't ned the SSR you can deactivate it in Nuxt

**Reply by nikkioon** [reply]
thats the problem i have disabled it 
<img width="447" alt="2022-05-19 um 11 21 51" src="https://user-images.githubusercontent.com/104486336/169260078-388ef01c-f5bd-4666-b8b0-b4025a11bfc1.png">
and still got the error

**Reply by posva** [reply]
Seems unlikely but in that case, it's probably a bug on Nuxt or h3

**Reply by tepafril** [reply]
I am also facing the same issue. Any solution? Please?
